### PR TITLE
GEODE-8964: Entry created via TX putIfAbsent with null value can write incorrect data to cache

### PR DESCRIPTION
**Pull request by luissson**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/internal/cache/TXCommitMessage.java [hunk 1]
```diff
@@ -754,8 +754,14 @@ void firePendingCallbacks(List<EntryEventImpl> callbacks) {
           ee.getRegion().invokeTXCallbacks(EnumListenerEvent.AFTER_CREATE, ee, true,
               isLastTransactionEvent);
         } else {
-          ee.getRegion().invokeTXCallbacks(EnumListenerEvent.AFTER_UPDATE, ee, true,
-              isLastTransactionEvent);
+          if (ee.getNewValue() == null) { // GEODE-8964, fixes GII and TX create conflict that
```
Comment (DonalEvans) [inline geode-core/src/main/java/org/apache/geode/internal/cache/TXCommitMessage.java:757]: Small nitpick, but could the comment here be put on its own line?

File: geode-core/src/test/java/org/apache/geode/internal/cache/TXCommitMessageTest.java [hunk 2]
```diff
@@ -80,4 +81,48 @@ public void firePendingCallbacksSetsChangeAppliedToCacheInEventLocalFilterInfo()
...
+
+    when(updateEvent.getLocalFilterInfo()).thenReturn(null);
+    when(updateEvent.getNewValue()).thenReturn(null);
+    when(updateEvent.getRegion()).thenReturn(region);
+
+    doReturn(lastTxEvent).when(message).getLastTransactionEvent(callbacks);
+
+    message.firePendingCallbacks(callbacks);
+
+    verify(region).invokeTXCallbacks(EnumListenerEvent.AFTER_CREATE, updateEvent, true, false);
```
Comment (DonalEvans) [inline geode-core/src/test/java/org/apache/geode/internal/cache/TXCommitMessageTest.java:103]: The two test cases here can be condensed to one if this line is changed to `verify(region, only()).invokeTXCallbacks(EnumListenerEvent.AFTER_CREATE, updateEvent, true, false);` which will validate both that `invokeTXCallbacks()` is called with the correct arguments and that it is *only* called with those arguments, removing the need for a separate test to confirm that it's not called with the `AFTER_UPDATE` argument.

**Review by DonalEvans — CHANGES_REQUESTED** [review]
Just a little code simplification/cleanup.

**Review by DonalEvans — APPROVED** [review]

**Comment by luissson** [comment]
only() was on the :nose:, thanks for the feedback!

**Review by kirklund — APPROVED** [review]
Looks great. Thank you!

**Pull request merged**